[Agent] Vector aggregation update with new config key (#13076)
DataDog/documentation: content/en/agent/vector_aggregation.md

**Notes**:

- Only logs and metrics aggregation is supported.
- Vector can directly collect logs and metrics from alternative sources. When doing so, third party logs may not include proper tagging. A convenient way to [add tags][8], source or service values is to use the [Vector Remap Language][9].

## Configuration

### Agent configuration
To send logs to Vector, update the Agent configuration file, `datadog.yaml`.

> For logs update the following values in the `datadog.yaml` file:

Reviewer: ```suggestion
For logs, update the following values in the `datadog.yaml` file:
```

```yaml
vector:
  logs.enabled: true
  # Adjust protocol to https if TLS/SSL is enabled on the Vector side
  logs.url: "http://<VECTOR_HOST>:<VECTOR_PORT>"
# Uncomment the following line if you use a version of Vector before v0.17.0
# logs_config.use_v2_api: false 
```

> For metrics update the following values in the `datadog.yaml` file:

Reviewer: ```suggestion
For metrics, update the following values in the `datadog.yaml` file:
```

If you are using Docker, add the following to your Agent configuration file.

```
-e DD_VECTOR_METRICS_URL=http://<VECTOR_HOST>:<VECTOR_PORT>
-e DD_VECTOR_METRICS_ENABLED=true
-e DD_VECTOR_LOGS_URL=http://<VECTOR_HOST>:<VECTOR_PORT>
-e DD_VECTOR_LOGS_ENABLED=true
```

### Vector configuration

> To receive logs or metrics from Datadog Agent, configure Vector with a [datadog_agent source][10]. To send logs to

Reviewer: ```suggestion
To receive logs or metrics from the Datadog Agent, configure Vector with a [datadog_agent source][10]. To send logs to
```

Datadog, Vector must be configured with at least one [datadog_logs sink][11]. Similarly to send metrics to Datadog,
Vector must be configured with at least one [datadog_metrics sink][12].

See the official [Vector documentation][13] for all available configuration parameters and transforms that can be
applied to logs while they are processed by Vector.

> Here is a configuration example that adds a tag to every log and avery metric using the Vector Remap Language:

Reviewer: ```suggestion
Here is a configuration example that adds a tag to every log and metric using the Vector Remap Language:
```

to the `agents.customAgentConfig` value. **Note**: `agent.useConfigMap` must be set to `true`
for `agents.customAgentConfig` to be taken into account.

For additional details about the Datadog Helm chart, see [the Kubernetes documentation][14].

Vector provides an [official chart for aggregating data][15] that comes with a Datadog
logs source preconfigured. For more information about installing Vector using Helm,
see to the [official Vector documentation][16].

> To send logs to Datadog, a `datadog_logs` sink need to be added to the Vector configuration. Similarly to send metrics

Reviewer: ```suggestion
To send logs to Datadog, a `datadog_logs` sink needs to be added to the Vector configuration. Similarly, to send metrics
```

> to Datadog a `datadog_metrics` sinks need to be added to Vector configuration. Vector's chart can hold any valid Vector

Reviewer: ```suggestion
to Datadog, a `datadog_metrics` sinks needs to be added to Vector configuration. Vector's chart can hold any valid Vector
```

> configuration in the `values.yaml` file using the `customConfig` field. To enable `datadog_logs` the same kind of

Reviewer: ```suggestion
configuration in the `values.yaml` file using the `customConfig` field. To enable `datadog_logs`, the same kind of
```

configuration as described under [Vector configuration](#vector-configuration) can be directly included as-is in the
Vector chart configuration.

## Manipulating Datadog logs and metrics with Vector

Logs and metrics sent to Vector can benefit from the full capabilities of Vector, including [Vector Remap Language][3]
for transformations.

When received by Vector, logs sent by the Datadog Agent are structured using the expected schema. When submitting logs

> directly to the Datadog API, see the [API documentation][17] for a complete schema description.

Reviewer: ```suggestion
using the Datadog API, see the [API documentation][17] for a complete schema description.
```

> Logs and metrics collected by Vector from other sources can be [fully enriched][8]. VRL can be used to adjust those logs

Reviewer: ```suggestion
Logs and metrics collected by Vector from other sources can be [fully enriched][8]. VRL can be used to adjust logs
```
